modules/analyze_basic_content_info/src/Plugin/Analyze/ContentInfo.php

```diff
@@ -0,0 +1,175 @@
+   *   The entity field manager.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   Entity type manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   The renderer service.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $languageManager
+   *   The language manager service.
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    HelperInterface $helper,
+    AccountProxyInterface $currentUser,
+    protected EntityFieldManagerInterface $entityFieldManager,
+    protected EntityTypeManagerInterface $entityTypeManager,
+    protected RendererInterface $renderer,
+    protected LanguageManagerInterface $languageManager
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $helper, $currentUser);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition): static {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('analyze.helper'),
+      $container->get('current_user'),
+      $container->get('entity_field.manager'),
+      $container->get('entity_type.manager'),
+      $container->get('renderer'),
+      $container->get('language_manager'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function renderSummary(EntityInterface $entity): array {
+    return [
+      '#type' => 'table',
+      '#header' => [['data' => 'Basic Info', 'colspan' => 2, 'class' => ['header']]],
+      '#rows' => [
+        ['data' => ['Word count', $this->getWordCount($entity)]],
+        ['data' => ['Image count', $this->getImageCount($entity)]],
+      ],
+      '#attributes' => [
+        'class' => ['basic-data-table'],
+        'style' => ['table-layout: fixed;'],
+      ],
+    ];
+  }
+
+  /**
+   * Helper to calculate a rough word count for the entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to calculate a word count for.
+   *
+   * @return int
+   *   The counted words. Defaults to 0.
+   */
+  private function getWordCount(EntityInterface $entity): int {
+    // Get the current active langcode from the site.
+    $langcode = $this->languageManager->getCurrentLanguage()->getId();
+
+    // Get the rendered entity view in default mode.
+    $view = $this->entityTypeManager->getViewBuilder($entity->getEntityTypeId())->view($entity, 'default', $langcode);
+    $rendered = $this->renderer->render($view);
+    return str_word_count(strip_tags($rendered->__toString()));
+  }
+
+  /**
+   * Helper to get a count of images attached to an entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to count images on.
+   *
+   * @return int
+   *   The number of images. Defaults to 0.
+   */
+  private function getImageCount(EntityInterface $entity): int {
```

Comment:
Possibly as a follow on we could switch to counting images in the rendered markup so paragraphs etc are included?

Reply:
This one and the text I think will need some more changes when Juurian looks at it, since he would know the specifics what should be counted here and in the text. So my suggestion is that we should add him to the review before we continue on this one.

Reply:
This is not really working for me, we'll need to just count images in rendered HTML. This does not work a couple of situations that are common in our clients' sites:

1. CKEditor or DXPR Builder is used to add images with external URL, i.e. end-user drops an absolute URL instead of using upload/media browser.
2. When using Media Browser with DXPR Builder pages, the image entity will not attach to the entity. We just never bothered doing that and never thought about it. offtopic: do you think that's bad?

In the future, when we implement our AI Assistant in DXPR CMS, we will use an API for generating placeholder images with text in them, these will also just be image tags with an external absolute URL have have no representation as entities in Drupal.